cob/cob.c

```diff
@@ -2171,6 +2171,7 @@ M0_INTERNAL int m0_cob_delete(struct m0_cob *cob, struct m0_be_tx *tx)
 		cob_table_delete(cob->co_dom->cd_fileattr_omg, tx, &key);
 	}
 out:
+	m0_cob_put(cob);
```

Comment:
m0_cob_delete()  should not call m0_cob_put(), instead callers of m0_cob_delete() should take care of it.
Otherwise, callers can use m0_cob_delete_put() instead m0_cob_delete().

Looking at the code, all the callers of m0_cob_delete() are taking care of m0_cob_put(), except cd_cob_delete(). 
so cd_cob_delete() should call m0_cob_delete_put() instead of m0_cob_delete()

Reply:
Updated.